datatables.go

```diff
@@ -374,7 +374,15 @@ func (p *Parser) bindGrenadeProjectiles(entity *st.Entity) {
 	})
 
 	entity.FindPropertyI("m_nModelIndex").OnUpdate(func(val st.PropertyValue) {
+
 		proj.Weapon = p.grenadeModelIndices[val.IntVal]
+		proj.WeaponInstance = common.NewEquipment(p.grenadeModelIndices[val.IntVal])
+
+		// We also need to store this Equipment in a map for this player
+		// Note: I guess OnUpdate() is asynchrone, so not sure if Owner is loaded at this stage
+		if proj.Owner != nil {
+			p.gameState.thrownGrenades[proj.Owner][proj.EntityID] = &proj.WeaponInstance
+		}
```

Comment:
I'm a bit sceptic about this block, but I will need to check it in more detail.

Reply:
Indeed, this goes along with `datatables.go` --> `entity.FindPropertyI("m_hOwnerEntity").OnUpdate` -> line 400 ...

I don't like it either but i didn't know how to deal with this async functions :-(

Reply:
Maybe there is a way to handle this part AFTER ALL entity properties are updated ?

Reply:
I think here would be perfect : 

```go
entity.OnCreateFinished(func() {

	if proj.Owner != nil {
		p.gameState.thrownGrenades[proj.Owner][proj.EntityID] = proj.WeaponInstance
	}

	p.eventDispatcher.Dispatch(events.GrenadeProjectileThrow{
		Projectile: proj,
	})
})
```

This way i could remove the other part i showed you!

Reply:
Yeah that would be much better 😄.

An no worries, I didn't even know myself out of the top of my head. Also that's why we can have these discussions on the PRs 🙂